File: src/栈和队列/406.根据身高重建队列/406.根据身高重建队列.py
# 先排序 再插队
def reconstructQueue(people):
    people = sorted(people, key = lambda x:(-x[0], x[1]))
    res = []
    for i in range(len(people)):
        resLen = len(res)
        if people[i][1] < resLen:
            res.insert(people[i][1], people[i])
        else:
            res.append(people[i])
    return res

people = [[7,0],[4,4],[7,1],[5,0],[6,1],[5,2]]
print(reconstructQueue(people))

# 精选解题方法
# https://leetcode.cn/problems/queue-reconstruction-by-height/solutions/486493/xian-pai-xu-zai-cha-dui-dong-hua-yan-shi-suan-fa-g/

File: src/栈和队列/406.根据身高重建队列/406.根据身高重建队列.js
/**
 * @param {number[][]} people
 * @return {number[][]}
 */
var reconstructQueue = function(people) {
    people.sort((a, b) => {
        if(a[0] === b[0]){
            return a[1] - b[1]
        } else {
            return b[0] - a[0]
        }
    })
    const res = Array();
    for(let i = 0; i < people.length; i++){
        const resLen = res.length;
        if(people[i][1] >= resLen){
            res.push(people[i])
        } else {
            res.splice(people[i][1], 0, people[i])
        }
    }
    return res;
};